packages/cli/src/workflow-execute-additional-data.ts

```diff
@@ -80,11 +80,20 @@ export function objectToError(errorObject: unknown, workflow: Workflow): Error {
 	if (errorObject instanceof Error) {
 		// If it's already an Error instance, return it as is.
 		return errorObject;
-	} else if (errorObject && typeof errorObject === 'object' && 'message' in errorObject) {
+	} else if (
+		isObjectLiteral(errorObject) &&
+		'message' in errorObject &&
+		typeof errorObject.message === 'string'
+	) {
```

Comment:
**question (non-blocking):** What are your thoughts about using zod or a class validator for this?

Reply:
My impression is `objectToError` is a patch for the fact that we cannot directly use the `ExecutionError` that's being thrown by a failed subworkflow execution. Ideally we should be able to trust our own errors - every error in the `ExecutionError` union should already have the requisite data, with no need for post-processing or validation.